Source/WebCore/accessibility/isolatedtree/AXIsolatedTree.cpp

```diff
@@ -1352,8 +1363,12 @@ void AXIsolatedTree::processQueuedNodeUpdates()
 #if ENABLE(AX_THREAD_TEXT_APIS)
 AXTextMarker AXIsolatedTree::firstMarker()
 {
-    RefPtr root = rootNode();
-    return root ? AXTextMarker { root->treeID(), root->objectID(), 0 } : AXTextMarker();
+    ASSERT(!isMainThread());
+    // The first marker should be constructed from the WebArea, not the ScrollView (the root),
```

Comment:
Maybe "The first marker should be constructed from the root WebArea, not the true root of the tree which is the ScrollView, so that..."

Take it or leave it, just a bit more precise that we are looking specifically for the root web area.

Reply:
Updated!